public/src/main/java/com/revenuecat/purchases/models/Period.kt

```diff
@@ -0,0 +1,71 @@
+package com.revenuecat.purchases.models
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Represents subscription or [PricingPhase] billing period
+ */
+@Parcelize
+data class Period(
+    /**
+     * The number of period units.
+     */
+    val value: Int,
+
+    /**
+     * The increment of time that a subscription period is specified in.
+     */
+    val unit: Unit,
+
+    /**
+     * Specified in ISO 8601 format. For example, P1W equates to one week,
+     * P1M equates to one month, P3M equates to three months, P6M equates to six months,
+     * and P1Y equates to one year
+     */
+    val iso8601: String
+) : Parcelable {
+    @SuppressWarnings("MagicNumber")
+    enum class Unit {
+        DAY,
+        WEEK,
+        MONTH,
+        YEAR,
+        UNKNOWN
+    }
+}
+
+// Would use Duration.parse but only available API 26 and up
+fun String.toPeriod(): Period {
```

Comment:
Might be a style question, but why don't we create a `Period(String iso8601Duration)` constructor instead? I find this a lot more difficult to find.

Reply:
I was following some of the other patterns here already (like for `RecurrenceMode`) but I kind of agree with this one 🤷‍♂️  I feel like this does make it easier to read/understand

Reply:
Or a static method if you want to support potential empty returns, but I think having it on `Period` not `String` like this extension would be easier to find (but again minor, feel free to ignore).